### PR TITLE
[Feature] Some metadata on QGF/QFF files

### PR DESCRIPTION
**Pull request by elpekenin**
## Description

As per @tzarc's [request](https://discord.com/channels/440868230475677696/1052728170236874772/1084294383857324043), some metadata on the files generated.

I also moved some duplicated code into a shared function. There's still a lot of duplication bewtween between fonts and images, but didn't feel like rewriting everything.

Removed a couple of `file.close()` along the way, as the context manager (aka `with ...` block) already closes them when we leave its scope.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [X] New feature
- [X] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by elpekenin** [comment]
Btw, i can happily add more metadata, just let me know what else we could/should have

File: lib/python/qmk/painter.py [hunk 1]
```diff
@@ -79,6 +80,110 @@
...
+
+
+def generate_subs(cli, out_bytes, *, font_metadata=None, image_metadata=None):
+    if font_metadata is not None and image_metadata is not None:
+        raise ValueError("Cant generate subs for font and image at the same time")
+
+    # Work out the command that was executed
+    args = [arg for arg in cli.args.__dict__ if not arg.startswith("_") and arg not in ("datetime_fmt", "log_fmt", "log_file_fmt", "log_file_level", "log_file", "color", "interactive", "config_file", "subparsers", "entrypoint")]
+    args_str = " ".join((f"--{arg} {getattr(cli.args, arg)}" for arg in sorted(args)))
+    command = f"qmk {cli.args.subparsers} {args_str}"
```
Comment (zvecr) [inline lib/python/qmk/painter.py:147]: This wouldnt be accepted, the context of what command was ran should just be passed through to the function

**Review by zvecr — CHANGES_REQUESTED** [review]

File: lib/python/qmk/painter.py (hunk 1 above)
Comment (elpekenin) [reply lib/python/qmk/painter.py:147]: I made this hacky way just so this updates itself when a command/arg is added/renamed.

Would this kind of code be accepted at the command's function (eg: `painter_convert_graphics`) or should it be hardcoded in?

**Comment by drashna** [comment]
> lib/python/qmk/painter_qgf.py:224:1: C901 '_save' is too complex (17)

**Comment by elpekenin** [comment]
> > lib/python/qmk/painter_qgf.py:224:1: C901 '_save' is too complex (17)


Not my fault, i promise. I added a couple extra lines to something that was barely out of warning threshold.

Btw, i made another PR afterwards to reduce the function into smaller ones. Still waiting for reviews too

**Comment by elpekenin** [comment]
Checking if a comment makes stalebot happy... This is not stale, but just waiting for reviews :)

**Comment by tzarc** [comment]
I'm assuming #21124 fixes the lint issue on this one, once rebased. On hold for that one.

**Review by tzarc — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**